var/spack/repos/builtin/packages/py-genshi/package.py

```diff
@@ -9,10 +9,10 @@
 class PyGenshi(PythonPackage):
     """Python toolkit for generation of output for the web"""
     homepage = "https://genshi.edgewall.org/"
-    url      = "http://ftp.edgewall.com/pub/genshi/Genshi-0.7.tar.gz"
+    url = "https://github.com/edgewall/genshi/archive/refs/tags/0.7.7.tar.gz"
+    git = "https://github.com/edgewall/genshi.git"
 
-    version('0.7', sha256='1d154402e68bc444a55bcac101f96cb4e59373100cc7a2da07fbf3e5cc5d7352')
-    version('0.6.1', sha256='fed947f11dbcb6792bb7161701ec3b9804055ad68c8af0ab4f0f9b25e9a18dbd')
-    version('0.6', sha256='32aaf76a03f88efa04143bf80700399e6d84eead818fdd19d763fd76af972a4b')
```

Comment:
Spack policy is to first deprecate versions before removing them: https://spack.readthedocs.io/en/latest/packaging_guide.html#deprecating-old-versions

Reply:
Ah, I was not familiar with the deprecation policy and that makes sense.  I saw runtime errors with the older Genshi versions, so I'll need to add a function to the spack package run their unittest suite.  Looking at it now I wonder whether the errors I saw earlier were related to the missing `py-six` dependency even though they did not appear to be import related.